src/Clients/Producer/KafkaProducer.php

```diff
@@ -13,11 +13,25 @@ class KafkaProducer extends Producer
 {
     private const RD_KAFKA_MSG_F_COPY = 0;
 
-    public function __construct(ProducerConfig $config)
+    /** @var bool */
+    private $flushOnDestruct;
```

Comment:
proč tahle option?

Reply:
budu tu chtit udelat ve vysledku asi 3 moznost. on destruct do nothing/flush/purge

Reply:
ok no, mě tohle přijde dost divný ... čekal bych že jak jednou zavolám produce, tak to bude v queue ... a tím bych tady nečekal nic jinýho než flush. Jaký máš důvody pro ty další možnosti?

Reply:
jo ale mas 2 (purge,flush (nothing neni, to je purge) moznosti, jak to ohandlovat. a musis to ohandlovat sam

Reply:
proč? :D Jednou jsem dal produce, tak čekám, že se to producene ... takže implicitní flush on destruct je imo cesta

Reply:
to že má knihovna nějakej interní stav by mě zajímat nemělo, neměla by mě nutit ho zvenku řešit

Reply:
ale nuti, takze se nechci snazit bejt chytrej a delat ze vim, co user potrebuje. nejsem grekas. prijde mi validni usecase, ze chces rychlej exit a nezajimaj te nonproduced msgs

https://github.com/arnaud-lb/php-rdkafka#proper-shutdown

Reply:
ok whatever, přijde mi to jako divnej design :D

Reply:
byly issues s ukoncovanim produceru, exposed flush a purge tohle to resi no